### DISCUSSION TITLE
Export Animation Clips/Symbols?

### DISCUSSION THREAD

**Discussion by yaknepper** [keiwando/toonsquid#61] [General]
Is there a way to export an animation clip/symbol to be imported and used in another project?

If not, this feature would be an awesome addition to an already awesome app!

Cheers

**Comment by Firefly65536** [comment]
I want this too!

**Reply by michaelwhotoon** [reply]
Doesn't make sense that you have to recreate the symbol in every new project. This really takes away the convenience of a symbol. I hardly every reuse a sequence (symbol) again in the same project but would use it in just about every new project (Mouth symbols)

**Reply by dabing1022** [reply]
Indeed, the symbol source file feels better than the sequence image or video

**Reply by MojoMan49** [reply]
Sure, because one can use the markers in the clip on the fly. Can’t do that w/ movie clips.

**Comment by Cat-Butler** [comment]
Symbols/animation clips can be exported/imported from the Library as videos or image sequences (but not yet, unfortunately, as vectors). I’ve found for my purposes it works best as MP4 videos (your mileage might vary).

Change to the Library, touch and hold on the clip/symbol you want to export, tap Export from the menu that appears across the bottom of the Library sub-window. Import into other projects as needed.

Animate using **time** keyframes (as per the handbook) for e.g. mouth symbols.

**Reply by DoodlerPete** [reply]
I would to be able to export a vector symbol and all its markers as an asset to be able to use in other projects. This feature would make me give up my other animation apps. I have a mouth setup I want to use on all of my characters in all of my projects. A great app otherwise. Thank you.

**Comment by MrMikeMuzik** [comment]
I too would love to Export and Import Animation Clips/Symbols as vectors.

**Reply by MojoMan49** [reply]
Agree, it would have seemed that making symbols universal to all projects would be  a great addition.

Repurposing mouths and eyes for instance would be really useful

![3690A513-D611-4E43-B89E-61E263D987EF](https://user-images.githubusercontent.com/127110520/225358816-32182932-f7c9-47ad-80de-ea80e9130172.jpeg)

**Comment by Lioncub901** [comment]
Yes, we need this. Maybe add a “move to project” button also.

**Comment by jordanwmartin** [comment]
Yeah, this has been mentioned before. I still hold that this missing feature is the only thing stopping me from using this professionally. I need to be able to reuse symbols, and in the best world, be able to import them from other applications in a standard format like SVG.

**Reply by MojoMan49** [reply]
I agree. This I believe would be a HUGE timesaver.

If our genius developer (not sarcastic here with the use of genius) wants to see a quantum leap in sales of the app, then us would be animators posting our “Cartoons” out their on YouTube and other social media would be the living testimonial.

As well, a better way to handle scenes and typical things like fades, curtains and iris  would be really cool.

BUT for me the highest priority are symbols in an app level library (probably will need sub folders down the road) would be fantastic

So, if I wanted to create a symbol out of “Jonah” the kid and use this as a basis in other scenes of projects, that would be wonderful


https://user-images.githubusercontent.com/127110520/227804135-4e169e75-880a-4209-8621-5e57272aae83.mp4

**Comment by DoodlerPete** [comment]
I have a work around. Create a project with all your assets/symbols in that you continually use. And make that your template for future projects. I use the same mouth and eyes for all my characters and this is working great so far. You just have to make sure that all the time markers are set first.
 I hope this helps. 
 
 This is my first animation. 
 

https://user-images.githubusercontent.com/98691740/235222330-6f50d3ff-5b29-4735-a98d-ae639dc14259.mov

**Reply by jordanwmartin** [reply]
That's a creative solution for sure, but it's still way too limited. Asset libraries change over time, and things get modified as you work. And for me, I need to be able to add older assets, or new assets from other applications. 

(fantastic animation by the way!)

**Reply by MojoMan49** [reply]
Would you be good enough to show the structure of your timelines and/or libraries.

If one is going to use the same characters, then this might work out.

**Comment by keiwando** [comment]
With the 1.9 update you can now export selections from a project's library into separate .tsproj files which you can then import into other projects.

**Reply by yaknepper** [reply]
Yes! What an awesome update. Thank you very much.

**Reply by Cat-Butler** [reply]
Yes, thank you Keiwan for adding this feature!
